src/rts/core/rev_tran.py

```diff
@@ -15,32 +15,254 @@
+            log.debug(
+                "Metadata for accession '%s' has not changed, skipping upsert.",
+                accession,
+            )
+            return
+
+        await self._metadata_dao.upsert(study_metadata)
+
+        log.debug("Transpiling metadata to workbook for accession '%s'.", accession)
+        workbook = self.reverse_transpile(study_metadata)
+
+        log.debug("Workbook created for accession '%s', upserting to DB.", accession)
+        await self._workbook_dao.upsert(workbook=workbook, study_accession=accession)
+
+    async def retrieve_metadata(self, *, study_accession: str) -> StudyMetadata:
+        """Retrieve study metadata from the DAO by its accession.
+
+        Raises MetadataNotFoundError if the metadata does not exist for the
+        given study accession.
+        """
+        try:
+            return await self._metadata_dao.get_by_id(study_accession)
+        except ResourceNotFoundError as err:
+            error = self.MetadataNotFoundError(study_accession=study_accession)
+            log.error(error)
+            raise error from err
+
+    async def delete_metadata(self, *, study_accession: str) -> None:
+        """Delete study metadata from the database by its accession.
+
+        This method will always try to delete the associated workbook as well,
+        regardless of whether the metadata exists or not.
+
+        Does not raise an error if the metadata or workbook does not exist.
+        """
+        try:
+            await self._metadata_dao.delete(study_accession)
+        except ResourceNotFoundError:
+            log.debug("Metadata for accession '%s' already deleted.", study_accession)
 
-    async def json_to_xlsx(
+        await self._workbook_dao.delete(study_accession=study_accession)
+        log.info("Workbook and metadata deleted for accession '%s'.", study_accession)
+
+    async def retrieve_workbook(self, *, study_accession: str) -> bytes:
+        """Retrieve the workbook for a given study accession.
+
+        Args:
+        - `study_accession`: The accession of the study metadata to retrieve.
+
+        Raises MetadataNotFoundError if the workbook does not exist.
+
+        Returns:
+        - The workbook as bytes.
+        """
+        try:
+            return await self._workbook_dao.find(study_accession=study_accession)
+        except ResourceNotFoundError as err:
+            error = self.MetadataNotFoundError(study_accession=study_accession)
+            log.error(error)
+            raise error from err
+
+    def _translate_sheet_name(self, sheet_name: str) -> str:
+        """Rename sheets in the workbook to with configured values.
+
+        Raises SheetNamingError if the sheet name is not configured.
+        """
+        if sheet_name not in self._config.sheet_names:
+            error = self.SheetNamingError(sheet_name=sheet_name)
+            log.error(error)
+            raise error
+        return self._config.sheet_names[sheet_name]
```

Comment:
```suggestion
        return self._config.sheet_names.get(sheet_name, sheet_name)
```
I would suggest to not make the mapping optional and just use the name "as is" when nothing is configured.

Reply:
I changed it to do that and also to truncate the original name to 31 characters to avoid Excel having to bother the user about it.